content/en/docs/tasks/network/extend-service-ip-ranges.md

```diff
@@ -0,0 +1,182 @@
+
+[![asciicast](https://asciinema.org/a/618594.svg)](https://asciinema.org/a/618594)
+
+### Adding a new ServiceCIDR
+
+On a cluster with a 10.96.0.0/28 range for Services, there is only 2^(32-28) - 2 = 14 IP addresses avaalable. The `kubernetes.default`  Service is always created, that leaves us with only 13 possible Services.
+
+```sh
+for i in $(seq 1 13); do kubectl create service clusterip "test-$i" --tcp 80 -o json | jq -r .spec.clusterIP; done
+```
+```
+10.96.0.11
+10.96.0.5
+10.96.0.12
+10.96.0.13
+10.96.0.14
+10.96.0.2
+10.96.0.3
+10.96.0.4
+10.96.0.6
+10.96.0.7
+10.96.0.8
+10.96.0.9
+error: failed to create ClusterIP service: Internal error occurred: failed to allocate a serviceIP: range is full
+```
+
+The number of available addresses can be increased by creating a new ServiceCIDR that extends or adds new IP ranges.
+
+```sh
+cat <EOF | kubectl apply -f -
+apiVersion: networking.k8s.io/v1alpha1
+kind: ServiceCIDR
+metadata:
+  name: newcidr1
+spec:
+  cidrs:
+  - 10.96.0.0/24
+EOF
+```
+```
+servicecidr.networking.k8s.io/newcidr1 created
+```
+
+and this will allow us to create new Services with ClusterIPs that will be picked from this new range. 
+
+```sh
+for i in $(seq 13 16); do kubectl create service clusterip "test-$i" --tcp 80 -o json | jq -r .spec.clusterIP; done
+```
+```
+10.96.0.48
+10.96.0.200
+10.96.0.121
+10.96.0.144
+```
+
+### Deleting a ServiceCIDR
+
+A ServiceCIDR can not be deleted if there are IPAddressess that depend on it
+
+```sh
+kubectl delete servicecidr newcidr1
+```
+```
+servicecidr.networking.k8s.io "newcidr1" deleted
+```
+
+this is managed by a controller that runs in the kube-controller-manager
+that participates on the deletion of the object, and removes
+the finalizer once it verifies it will not leave orphan IPAddresses.
+
+```sh
+kubectl get servicecidr newcidr1 -o yaml
+```
+```
+apiVersion: networking.k8s.io/v1alpha1
+kind: ServiceCIDR
+metadata:
+  creationTimestamp: "2023-10-12T15:11:07Z"
+  deletionGracePeriodSeconds: 0
+  deletionTimestamp: "2023-10-12T15:12:45Z"
+  finalizers:
+  - networking.k8s.io/service-cidr-finalizer
+  name: newcidr1
+  resourceVersion: "1133"
+  uid: 5ffd8afe-c78f-4e60-ae76-cec448a8af40
+spec:
+  cidrs:
+  - 10.96.0.0/24
+status:
+  conditions:
+  - lastTransitionTime: "2023-10-12T15:12:45Z"
+    message: There are still IPAddresses referencing the ServiceCIDR, please remove
+      them or create a new ServiceCIDR
+    reason: OrphanIPAddress
+    status: "False"
+    type: Ready
+```
+
+
+By removing the Services containing the IP addresses that are blocking the deletion of the ServiceCIDR
+
+```sh
+for i in $(seq 13 16); do kubectl delete service "test-$i" ; done
+```
+```
+service "test-13" deleted
+service "test-14" deleted
+service "test-15" deleted
+service "test-16" deleted
+```
+
+the controller will remove the finalizer and the ServiceCIDR will be finally deleted.
```

Comment:
```suggestion
the control plane notices the removal. The control plane then removes its finalizer,
so that the ServiceCIDR that was pending deletion will actually be removed.
```